src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,257 @@
+﻿#if __ANDROID__
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Windows.UI.Notifications
+{
+	public partial class ToastNotifier
+	{
+		private string _channelID = "PlatformUno"; 
+		private Android.App.NotificationManager _notificationManager;
+
+		public ToastNotifier()
+		{
+			_notificationManager = Android.App.NotificationManager.FromContext(Android.App.Application.Context);
+
+			if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
+			{	// none of this strings would be visible for user
+				var channel = new Android.App.NotificationChannel(_channelID, "PlatformUnoChannel", Android.App.NotificationImportance.Default); ; // deprecated: Android.App.NotificationManager.ImportanceDefault
+				_notificationManager.CreateNotificationChannel(channel);
+			}
+
+		}
+
+		private string toastText2String(string toasttext)
+		{
+			// "When using the "ms-resource" prefix, the string identifier is referenced in the app's Resources.resw file."
+			if (!toasttext.StartsWith("ms-resource:"))
+			{
+				return toasttext;
+			}
+
+			toasttext = toasttext.Substring(12); // skipping prefix "ms-resource:"
+			var retVal = Windows.ApplicationModel.Resources.ResourceLoader.GetForCurrentView()?.GetString(toasttext);
+			if(retVal is null)
+			{	// we have no such string
+				return toasttext;
+			}
+			return retVal;
+		}
+
+		public void Show(ToastNotification notification)

```

Comment:
This method is too large, could you split it in smaller pieces to make it more readable ?

Reply:
You closed this comment without fixing it, please split this method in small parts.

Reply:
I closed it because `GetIconId` method was created from part of `Show`. One of two longer, complicated setters - other is text setters - I thought that copying objects are not good, but OK. I will extract also it.